libhfcommon/util.c

```diff
@@ -228,14 +229,19 @@ static __thread uint64_t       rndState[2];
 
 static void util_rndInitThread(void) {
 #if !defined(BSD)
-    int fd = TEMP_FAILURE_RETRY(open("/dev/urandom", O_RDONLY | O_CLOEXEC));
-    if (fd == -1) {
-        PLOG_F("Couldn't open /dev/urandom for reading");
-    }
-    if (files_readFromFd(fd, (uint8_t*)rndState, sizeof(rndState)) != sizeof(rndState)) {
-        PLOG_F("Couldn't read '%zu' bytes from /dev/urandom", sizeof(rndState));
+    size_t rrnd = 0;
+    size_t trnd = sizeof(rndState);
+
+    while (rrnd < trnd) {
+        // we are taking the harder road to cope with older linux versions without the getrandom's
+        // wrapper
+        ssize_t len = (ssize_t)syscall(SYS_getrandom, (uint8_t*)rndState + rrnd, trnd - rrnd, 0);
```

Comment:
1. Handle EINTR via TEMP_FAILURE_RETRY()
2. syscall() is a vararg, you need to cast all arguments to (long) becasue badness can happen in some cases.

Use can use files_readFromFd from libhfcommon/files.c as an example of TEMP_FAILURE_RETRY use.